### DISCUSSION TITLE
Simpler lighter weight ES6 javascript template

### DISCUSSION THREAD

**Discussion by steowens** [wailsapp/wails#2226] [Ideas]
I would like to see a Node.js free template which only uses ES6 javascript modules and html for the front end.  No binding and a way to enable menus to be created attached to the main window which call back to go.  The go would then decide what assets to navigate to and those could be resolved without needing to run a bunch of transformations ont them.

Is this possible, I tried gutting a project generated by the wails init command but when I did a wails build it died looking for package.json.

**Comment by leaanthony** [comment]
You can just provide an embed.FS with whatever you want in it. You may want to also update your `wails.json` to change the npm calls on build.

**Comment by steowens** [comment]
Thanks for the tip!  This is great, things are working pretty well.  I have a couple more bonus questions. 

 I notice that the build script generates icons but I want to provide my own icons instead.

 I would also like to modify the window title and make it something other than the project name.  Finally I would like to be able to open up a javascript debugger like the ones we have in regular browsers.  

Are any of these things possible and if so how can I make them happen?  Also is it preferable to open separate issues for these additional questions?

**Reply by leaanthony** [reply]
`wails dev` or `wails build -debug` will give you the dev tools on right click. Replace the icon in `build/` should work but some have recently reported it not picking it up (will check). https://wails.io/docs/reference/runtime/window#windowsettitle. Take your time to peruse the docs 👍

**Reply by steowens** [reply]
Yea the docs are pretty good and I seem to recall reding about there being a debug ability but when I went back looking for that reference I could not for the life find it.  I think it's probably important enough to get honorable mention in the main navigation area for the documentation.

**Reply by steowens** [reply]
Debug is working like a charm and I am quickly falling in love with this project.

**Reply by leaanthony** [reply]
Ooof why doesn't debug come up properly in the algolia search :/
